data/src/main/java/com/linkedin/data/schema/AbstractSchemaParser.java

```diff
@@ -0,0 +1,153 @@
+/*
+   Copyright (c) 2012 LinkedIn Corp.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package com.linkedin.data.schema;
+
+
+import com.linkedin.data.Data;
+import com.linkedin.data.DataList;
+import com.linkedin.data.DataMap;
+import com.linkedin.data.codec.DataLocation;
+import com.linkedin.data.codec.JacksonDataCodec;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Reader;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.IdentityHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+
+/**
+ * Common base class for parsers that parse Schema objects.
+ *
+ * @author slim
+ * @author jbetz
+ */
+abstract public class AbstractSchemaParser extends AbstractDataParser
```

Comment:
This code is has simply been refactored up from `SchemaParser` and is otherwise unchanged (except the constructor and `addTopLevelSchema()` method).